src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  useContext,
+  useCallback,
+  useMemo,
+  useEffect,
+  useState,
+  createContext,
+} from "react";
+import Swal from "sweetalert2";
+import withReactContent from "sweetalert2-react-content";
+import api from "../services/api";
+
+export interface AuthContextData {
+  signed: boolean;
+  register: (
+    name: string,
+    email: string,
+    password: string,
+    passwordConfirmation: string
+  ) => Promise<void>;
+  login: (email: string, password: string) => Promise<void>;
+  logout: () => Promise<void>;
+}
+
```

Comment:
I don't know about good practices in react, but wouldn't it be interesting to create a directory just for the interfaces?

Reply:
I'll make a research